vault/logical_system_activity_write_testonly.go

```diff
@@ -51,3 +56,133 @@ func (b *SystemBackend) handleActivityWriteData(ctx context.Context, request *lo
 	}
 	return nil, nil
 }
+
+// singleMonthActivityClients holds a single month's client IDs, in the order they were seen
+type singleMonthActivityClients struct {
+	// clients are indexed by ID
+	clients []string
+	// allClients contains all clients from all months
+	allClients map[string]*activity.EntityRecord
+}
+
+// multipleMonthsActivityClients holds multiple month's data
+type multipleMonthsActivityClients struct {
+	// months are in order, with month 0 being the current month and index 1 being 1 month ago
+	months     []*singleMonthActivityClients
+	allClients map[string]*activity.EntityRecord
+}
+
+// addNewClients generates clients according to the given parameters, and adds them to the month
+// if the namespace is empty, the client will be added to the defaultNamespace
+// the client will always have the actualMount as its mount accessor
+func (m *singleMonthActivityClients) addNewClients(c *generation.Client, defaultNamespace, actualMount string) error {
```

Comment:
Nit: ❓ Should we rename "actualMount" as "mountAccessor" as that would be more obvious/readable?